sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusException.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.exception.AmqpErrorCondition;
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.exception.AzureException;
+
+/**
+ * Defines {@link ServiceBusException} which has additional information about the operation that caused the
+ * error.
+ * @see ServiceBusErrorSource
+ */
+public final class ServiceBusException extends AzureException {
+    private final transient ServiceBusErrorSource errorSource;
+    private final transient ServiceBusFailureReason reason;
+    private final boolean isTransient;
+
+    /**
+     * @param throwable for the error happened.
+     * @param errorSource indicating which api caused the error.
+     */
+    public ServiceBusException(Throwable throwable, ServiceBusErrorSource errorSource) {
```

Comment:
Do we need the constructor to be public?

Reply:
Nope, made package private.